core/src/main/java/io/questdb/log/LogAlertSocket.java

```diff
@@ -0,0 +1,325 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.log;
+
+import io.questdb.VisibleForTesting;
+import io.questdb.network.Net;
+import io.questdb.std.*;
+
+import java.io.Closeable;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.function.Consumer;
+
+public class LogAlertSocket implements Closeable {
+
+    public static final String localHostIp;
+
+    static {
+        try {
+            localHostIp = InetAddress.getLocalHost().getHostAddress();
+        } catch (UnknownHostException e) {
+            throw new LogError("Cannot access our ip address info");
+        }
+    }
+
+    public static final String DEFAULT_HOST = "127.0.0.1";
+    public static final int DEFAULT_PORT = 9093;
+    public static final int IN_BUFFER_SIZE = 2 * 1024 * 1024;
+    public static final int OUT_BUFFER_SIZE = 4 * 1024 * 1024;
+    private static final int HOSTS_LIMIT = 12; // will never happen
+    private static final int FAIL_OVER_LIMIT = 100;
+
+
+    private final Rnd rand;
+    private final FilesFacade ff;
+    private final String[] hosts = new String[HOSTS_LIMIT];
+    private final int[] ports = new int[HOSTS_LIMIT];
+    private int hostPortSize;
```

Comment:
the name is hard to understand, perhaps a name describing the meaning would help

Reply:
the irony